docs/provision.md

```diff
@@ -0,0 +1,53 @@
+# Katello provisioning setup
+
+A role is included which will configure an all-on-one provisioning environment
+with Katello. It uses nested libvirt to make your virtual machine a hypervisor
+capable of running it's own VM's.  It also sets up a private NAT network on the
+host.
+
+## How to configure
+
+1. Enable nested virt on your phyiscal box.  See http://www.rdoxenham.com/?p=275
+comments for more details.  Essentially, add this line to kvm-intel.conf and
+reboot:
+
+        echo "options kvm-intel nested=1" | sudo tee /etc/modprobe.d/kvm-intel.conf
+
+2. Build a box
+
+    - Option 1: Use the `centos7-provisioning-nightly` box.
+
+    - Option 2: Use an existing katello box (e.g. centos7-katello-p4-nightly) and run the provisioning playbook (it takes a while, as it syncs (on-demand) centos7, puppet 4).   ***Note*: If you are using puppet 4, you need to increase the ram on the box, to something like 8096 otherwise candlepin crashes with OOM.**
+        `ansible-playbook -l centos7-katello-p4-nightly playbooks/katello_provisioning.yml`
+
+4. Login and create a compute profile, because this isn't possible with hammer or the API.
+
+   - Click Infrastructure / Compute Resources
+   - Click "libvirt"
+   - Click Compute profiles
+   - Click 2-Medium
+        - increse ram to 1024MB (required for centos 7)
+        - change network type to NAT, network name = provision
+   - Click Submit
+
+5.  Configure Activation Key
+
+  - Content/ Activation Keys
+  - Assign all available subscriptions to the activation key
```

Comment:
It is possible, but it's a mess of grepping and fact registration, since you need the UUID.  I can save it for a future enhancement...